File: ccif/src/java/other/org/himalay/ccif/ECM_statusReport.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:04 EST 2016")

public  class ECM_statusReport extends BinStruct implements PublicBinMsg{




    // members variables
    // downstreamCenterFreq
    public int downstreamCenterFreq ;
    // downstreamPowerLevel
    public int downstreamPowerLevel ;
    // bf1
    public BitField_8 bf1 ;
    // upstreamXmtCenterFreq
    public int upstreamXmtCenterFreq ;
    // upstreamPowerLevel
    public int upstreamPowerLevel ;
    // upstreamSymbolRate
    public short upstreamSymbolRate ;

        public ECM_statusReport () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize downstreamCenterFreq
        
        // Initialize downstreamPowerLevel
        
        // Initialize bf1
        bf1 = new BitField_8();
        // Initialize upstreamXmtCenterFreq
        
        // Initialize upstreamPowerLevel
        
        // Initialize upstreamSymbolRate
        
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read downstreamCenterFreq
        {downstreamCenterFreq=istream.readUnsignedShort(); retVal+=2;}
        // read downstreamPowerLevel
        {downstreamPowerLevel=istream.readUnsignedShort(); retVal+=2;}
        // read bf1
        retVal += bf1.read(istream);
        // read upstreamXmtCenterFreq
        {upstreamXmtCenterFreq=istream.readUnsignedShort(); retVal+=2;}
        // read upstreamPowerLevel
        {upstreamPowerLevel=istream.readUnsignedShort(); retVal+=2;}
        // read upstreamSymbolRate
        {upstreamSymbolRate=(short)(istream.readUnsignedByte()); retVal+=1;}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        
        
        
        
        
    
                // write downstreamCenterFreq
        ostream.writeShort(downstreamCenterFreq); retVal +=2;
        // write downstreamPowerLevel
        ostream.writeShort(downstreamPowerLevel); retVal +=2;
        // write bf1
        ostream.writeByte(bf1.getValue()); retVal +=1;
        // write upstreamXmtCenterFreq
        ostream.writeShort(upstreamXmtCenterFreq); retVal +=2;
        // write upstreamPowerLevel
        ostream.writeShort(upstreamPowerLevel); retVal +=2;
        // write upstreamSymbolRate
        ostream.writeByte(upstreamSymbolRate); retVal +=1;
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("ECM_statusReport\n");
    dc.increaseIndent();
        int retVal= 0;
        // write downstreamCenterFreq
        dc.indent();dc.getPs().println("downstreamCenterFreq="+downstreamCenterFreq+"(0x"+ Integer.toHexString(downstreamCenterFreq) + ")" );
        // write downstreamPowerLevel
        dc.indent();dc.getPs().println("downstreamPowerLevel="+downstreamPowerLevel+"(0x"+ Integer.toHexString(downstreamPowerLevel) + ")" );
        // write bf1
        {dc.indent();dc.getPs().print("downstreamCarrierLockStatus: ");dc.getPs().println(BitField_8.toDisplayString(getDownstreamCarrierLockStatus(),1));dc.indent();dc.getPs().print("reserved: ");dc.getPs().println(BitField_8.toDisplayString(getReserved(),2));dc.indent();dc.getPs().print("channel_s_cdmaStatus: ");dc.getPs().println(BitField_8.toDisplayString(getChannel_s_cdmaStatus(),2));dc.indent();dc.getPs().print("upstreamModulationType: ");dc.getPs().println(BitField_8.toDisplayString(getUpstreamModulationType(),3));}
        // write upstreamXmtCenterFreq
        dc.indent();dc.getPs().println("upstreamXmtCenterFreq="+upstreamXmtCenterFreq+"(0x"+ Integer.toHexString(upstreamXmtCenterFreq) + ")" );
        // write upstreamPowerLevel
        dc.indent();dc.getPs().println("upstreamPowerLevel="+upstreamPowerLevel+"(0x"+ Integer.toHexString(upstreamPowerLevel) + ")" );
        // write upstreamSymbolRate
        dc.indent();dc.getPs().println("upstreamSymbolRate="+upstreamSymbolRate+"(0x"+ Integer.toHexString(upstreamSymbolRate) + ")" );
dc.decreaseIndent();
        return retVal;
    }


        // Getter for downstreamCenterFreq
    //public int getDownstreamCenterFreq()
    //{
    //    return downstreamCenterFreq ;
    //}

    
    // Setter for downstreamCenterFreq
    //public void setDownstreamCenterFreq(int val)
    //{
    //    this.downstreamCenterFreq= val;
    //}
    // Getter for downstreamPowerLevel
    //public int getDownstreamPowerLevel()
    //{
    //    return downstreamPowerLevel ;
    //}

    
    // Setter for downstreamPowerLevel
    //public void setDownstreamPowerLevel(int val)
    //{
    //    this.downstreamPowerLevel= val;
    //}
    // Getter for bf1
    //public BitField_8 getBf1()
    //{
    //    return bf1 ;
    //}

    
    // Setter for bf1
    //public void setBf1(BitField_8 val)
    //{
    //    this.bf1= val;
    //}
    // Getter for upstreamXmtCenterFreq
    //public int getUpstreamXmtCenterFreq()
    //{
    //    return upstreamXmtCenterFreq ;
    //}

    
    // Setter for upstreamXmtCenterFreq
    //public void setUpstreamXmtCenterFreq(int val)
    //{
    //    this.upstreamXmtCenterFreq= val;
    //}
    // Getter for upstreamPowerLevel
    //public int getUpstreamPowerLevel()
    //{
    //    return upstreamPowerLevel ;
    //}

    
    // Setter for upstreamPowerLevel
    //public void setUpstreamPowerLevel(int val)
    //{
    //    this.upstreamPowerLevel= val;
    //}
    // Getter for upstreamSymbolRate
    //public short getUpstreamSymbolRate()
    //{
    //    return upstreamSymbolRate ;
    //}

    
    // Setter for upstreamSymbolRate
    //public void setUpstreamSymbolRate(short val)
    //{
    //    this.upstreamSymbolRate= val;
    //}


    public int getDownstreamCarrierLockStatus( )
    {
        return ( bf1.getValue() & 0x00000080 ) >> 7 ;
    }
    
    
    public void setDownstreamCarrierLockStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffff7f) | ( (val << 7 ) & 0x00000080));
    }
    
    
    public int getReserved( )
    {
        return ( bf1.getValue() & 0x00000060 ) >> 5 ;
    }
    
    
    public void setReserved(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffff9f) | ( (val << 5 ) & 0x00000060));
    }
    
    
    public int getChannel_s_cdmaStatus( )
    {
        return ( bf1.getValue() & 0x00000018 ) >> 3 ;
    }
    
    
    public void setChannel_s_cdmaStatus(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffffe7) | ( (val << 3 ) & 0x00000018));
    }
    
    
    public int getUpstreamModulationType( )
    {
        return ( bf1.getValue() & 0x00000007 ) >> 0 ;
    }
    
    
    public void setUpstreamModulationType(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xfffffff8) | ( (val << 0 ) & 0x00000007));
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/ECM_statusReport.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:16 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "ECM_statusReport.h"

#ifdef org_himalay_ccif_ECM_statusReport__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
ECM_statusReport::ECM_statusReport() {
    references= 0;


    // downstreamCenterFreq
    downstreamCenterFreq= (ui16)0;
    // downstreamPowerLevel
    downstreamPowerLevel= (ui16)0;
    // bf1
    bf1 = 0;
    // upstreamXmtCenterFreq
    upstreamXmtCenterFreq= (ui16)0;
    // upstreamPowerLevel
    upstreamPowerLevel= (ui16)0;
    // upstreamSymbolRate
    upstreamSymbolRate= (ui8)0;
}
// Destructor
ECM_statusReport::~ECM_statusReport() {
    // downstreamCenterFreq

    // downstreamPowerLevel

    // bf1

    // upstreamXmtCenterFreq

    // upstreamPowerLevel

    // upstreamSymbolRate

}

int ECM_statusReport::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // downstreamCenterFreq
    {
        downstreamCenterFreq=istream.readUI16();
        retVal+=2;
    }
    // downstreamPowerLevel
    {
        downstreamPowerLevel=istream.readUI16();
        retVal+=2;
    }
    // bf1
    bf1= istream.readUI8();
    retVal += 1;
    // upstreamXmtCenterFreq
    {
        upstreamXmtCenterFreq=istream.readUI16();
        retVal+=2;
    }
    // upstreamPowerLevel
    {
        upstreamPowerLevel=istream.readUI16();
        retVal+=2;
    }
    // upstreamSymbolRate
    {
        upstreamSymbolRate=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int ECM_statusReport::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read downstreamCenterFreq
    {
        downstreamCenterFreq=istream.readUI16();
        retVal+=2;
    }
    // read downstreamPowerLevel
    {
        downstreamPowerLevel=istream.readUI16();
        retVal+=2;
    }
    // read bf1
    bf1= istream.readUI8();
    retVal += 1;
    // read upstreamXmtCenterFreq
    {
        upstreamXmtCenterFreq=istream.readUI16();
        retVal+=2;
    }
    // read upstreamPowerLevel
    {
        upstreamPowerLevel=istream.readUI16();
        retVal+=2;
    }
    // read upstreamSymbolRate
    {
        upstreamSymbolRate=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int ECM_statusReport::write(DataOutputStream& ostream) { // throws IOException







    int retVal= 0;
    // write downstreamCenterFreq
    ostream.writeUI16(downstreamCenterFreq);
    retVal +=2;
    // write downstreamPowerLevel
    ostream.writeUI16(downstreamPowerLevel);
    retVal +=2;
    // write bf1
    ostream.writeUI8(bf1);
    retVal +=1;
    // write upstreamXmtCenterFreq
    ostream.writeUI16(upstreamXmtCenterFreq);
    retVal +=2;
    // write upstreamPowerLevel
    ostream.writeUI16(upstreamPowerLevel);
    retVal +=2;
    // write upstreamSymbolRate
    ostream.writeUI8(upstreamSymbolRate);
    retVal +=1;

    return retVal;
}

int ECM_statusReport::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("ECM_statusReport\n");
    dc.increaseIndent();
    int retVal= 0;
    // write downstreamCenterFreq
    dc.indent();
    dc.getPs().print("downstreamCenterFreq=");
    dc.getPs().println((long)downstreamCenterFreq);
    // write downstreamPowerLevel
    dc.indent();
    dc.getPs().print("downstreamPowerLevel=");
    dc.getPs().println((long)downstreamPowerLevel);
    // write bf1
    {
        dc.indent();
        dc.getPs().print("downstreamCarrierLockStatus: ");
        dc.getPs().printlnBin(getDownstreamCarrierLockStatus(),1);
        dc.indent();
        dc.getPs().print("reserved: ");
        dc.getPs().printlnBin(getReserved(),2);
        dc.indent();
        dc.getPs().print("channel_s_cdmaStatus: ");
        dc.getPs().printlnBin(getChannel_s_cdmaStatus(),2);
        dc.indent();
        dc.getPs().print("upstreamModulationType: ");
        dc.getPs().printlnBin(getUpstreamModulationType(),3);
    }
    // write upstreamXmtCenterFreq
    dc.indent();
    dc.getPs().print("upstreamXmtCenterFreq=");
    dc.getPs().println((long)upstreamXmtCenterFreq);
    // write upstreamPowerLevel
    dc.indent();
    dc.getPs().print("upstreamPowerLevel=");
    dc.getPs().println((long)upstreamPowerLevel);
    // write upstreamSymbolRate
    dc.indent();
    dc.getPs().print("upstreamSymbolRate=");
    dc.getPs().println((long)upstreamSymbolRate);
    dc.decreaseIndent();
    return retVal;
}


// Getter for downstreamCenterFreq
//ui16 ECM_statusReport::getDownstreamCenterFreq()
//{
//return this->downstreamCenterFreq;
//}

// Setter for downstreamCenterFreq
//void ECM_statusReport::setDownstreamCenterFreq(ui16 val)
//{
//this->downstreamCenterFreq= val;
//}

// Getter for downstreamPowerLevel
//ui16 ECM_statusReport::getDownstreamPowerLevel()
//{
//return this->downstreamPowerLevel;
//}

// Setter for downstreamPowerLevel
//void ECM_statusReport::setDownstreamPowerLevel(ui16 val)
//{
//this->downstreamPowerLevel= val;
//}

// Getter for bf1
//ui8 ECM_statusReport::getBf1()
//{
//return this->bf1;
//}

// Setter for bf1
//void ECM_statusReport::setBf1(ui8 val)
//{
//this->bf1= val;
//}

// Getter for upstreamXmtCenterFreq
//ui16 ECM_statusReport::getUpstreamXmtCenterFreq()
//{
//return this->upstreamXmtCenterFreq;
//}

// Setter for upstreamXmtCenterFreq
//void ECM_statusReport::setUpstreamXmtCenterFreq(ui16 val)
//{
//this->upstreamXmtCenterFreq= val;
//}

// Getter for upstreamPowerLevel
//ui16 ECM_statusReport::getUpstreamPowerLevel()
//{
//return this->upstreamPowerLevel;
//}

// Setter for upstreamPowerLevel
//void ECM_statusReport::setUpstreamPowerLevel(ui16 val)
//{
//this->upstreamPowerLevel= val;
//}

// Getter for upstreamSymbolRate
//ui8 ECM_statusReport::getUpstreamSymbolRate()
//{
//return this->upstreamSymbolRate;
//}

// Setter for upstreamSymbolRate
//void ECM_statusReport::setUpstreamSymbolRate(ui8 val)
//{
//this->upstreamSymbolRate= val;
//}

int ECM_statusReport::getDownstreamCarrierLockStatus( ) {
    return ( this->bf1 & 0x00000080 ) >> 7 ;
}

void ECM_statusReport::setDownstreamCarrierLockStatus(int val) {
    bf1=(  (this->bf1 & 0xffffff7f) | ( (val << 7 ) & 0x00000080));
}

int ECM_statusReport::getReserved( ) {
    return ( this->bf1 & 0x00000060 ) >> 5 ;
}

void ECM_statusReport::setReserved(int val) {
    bf1=(  (this->bf1 & 0xffffff9f) | ( (val << 5 ) & 0x00000060));
}

int ECM_statusReport::getChannel_s_cdmaStatus( ) {
    return ( this->bf1 & 0x00000018 ) >> 3 ;
}

void ECM_statusReport::setChannel_s_cdmaStatus(int val) {
    bf1=(  (this->bf1 & 0xffffffe7) | ( (val << 3 ) & 0x00000018));
}

int ECM_statusReport::getUpstreamModulationType( ) {
    return ( this->bf1 & 0x00000007 ) >> 0 ;
}

void ECM_statusReport::setUpstreamModulationType(int val) {
    bf1=(  (this->bf1 & 0xfffffff8) | ( (val << 0 ) & 0x00000007));
}

int ECM_statusReport::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_ECM_statusReport__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::ECM_statusReport* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::ECM_statusReport* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code